pkg/tests/end_to_end_tests/series_cache_test.go

```diff
@@ -0,0 +1,58 @@
+package end_to_end_tests
+
+import (
+	"context"
+	"sort"
+	"testing"
+
+	"github.com/jackc/pgx/v4/pgxpool"
+	"github.com/stretchr/testify/require"
+
+	"github.com/timescale/promscale/pkg/pgmodel/cache"
+	ingstr "github.com/timescale/promscale/pkg/pgmodel/ingestor"
+	"github.com/timescale/promscale/pkg/pgxconn"
+)
+
+func TestSeriesCacheRefreshOnStaleSeries(t *testing.T) {
+	withDB(t, *testDatabase, func(db *pgxpool.Pool, t testing.TB) {
+		conn := pgxconn.NewPgxConn(db)
+		ingestor, err := ingstr.NewPgxIngestorForTests(conn, nil)
+		require.NoError(t, err)
+		defer ingestor.Close()
+
+		ts := copyMetrics(generateLargeTimeseries())
+		samples, metadata, err := ingestor.Ingest(context.Background(), newWriteRequestWithTs(ts))
+		require.NoError(t, err)
+		require.NotEqual(t, 0, samples)
+		require.NotEqual(t, 0, metadata)
+
+		sCache := ingestor.SeriesCache().(*cache.SeriesCacheImpl)
+		underlyingCache := sCache.Internal()
```

Comment:
I would rather use the series cache to verify the series are present instead of adding public interface for underlying cache implementation just for tests.